enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/server/AuthServer.java

```diff
@@ -179,4 +180,20 @@ private static Server initServer() throws SSLException {
                 .sslContext(TLSUtils.buildGRPCServerSSLContext())
                 .build();
     }
+
+    private static void handleShutdown(Server grpcServer, RestServer restServer) {
+        // shutdown hook to handle server shutdown
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            logger.info("Received SIGTERM. Shutting down the server.");
+            grpcServer.shutdown();
+            try {
+                grpcServer.awaitTermination();
+            } catch (InterruptedException e) {
+                logger.error("Error while shutting down the gRPC server.", e);
+            }
+
+            restServer.shutDown();
+            logger.info("Server shut down.");
```

Comment:
```suggestion
            logger.info("Enforcer auth server shut down completed successfully");
```

Reply:
@renuka-fernando Shall we improve the error log mentioning `Enforcer` ?

Reply:
```suggestion
            logger.info("Enforcer shut down completed.");
```